woocommerce/woocommerce-ios
# Set transformer to `NSSecureUnarchiveFromData` for model versions 20-28 to fix warnings on iOS 13

Fixes #1372 

## Changes

Set the transformer to `NSSecureUnarchiveFromData` for `Transformable` attributes for model versions 20-28. For model versions below 20, the diffs are crazy large probably due to different Xcode versions so I didn't 

- Property ‘options’ on Entity ‘ProductAttribute’
- Property ‘crossSellIDs’ on Entity ‘Product’
- Property ‘groupedProducts’ on Entity ‘Product’
- Property ‘relatedIDs’ on Entity ‘Product’
- Property ‘upsellIDs’ on Entity ‘Product’
- Property ‘variations’ on Entity ‘Product’

## Testing

Please test migration from an earlier model version on an iOS 13 simulator/device, the earlier the better but not earlier than release 2.8.0.0 (model version 20), and make sure no warnings show up in the console about the transformer like in #1372. However, I saw a separate warning that I couldn't find a concrete fix for yet 😞

```
CoreData: annotation:  Failed to load optimized model at path '/Users/jaclynchen/Library/Developer/Xcode/DerivedData/WooCommerce-gyxkpbqewsthczdmcrmapoocxjsd/Build/Products/Debug-iphonesimulator/Storage.framework/WooCommerce.momd/Model 28.omo'
```

I've tested migrating from [release 3.9](https://github.com/woocommerce/woocommerce-ios/releases/tag/3.9) and [release 2.8](https://github.com/woocommerce/woocommerce-ios/releases/tag/2.8).

To build release 2.8:
- run `g checkout de299224ecddd3f7c7011e0a65325700db5706dd`
- run `bundle exec pod install`
- build the app from Xcode 11.1 (Zendesk used to be very sensitive to the Xcode version) onto an iOS 13.1 simulator

Update release notes:
